frontend/pages/sign-in/code.tsx

```diff
@@ -0,0 +1,192 @@
+import { ChangeEvent, ClipboardEvent, useCallback, useEffect } from 'react';
+
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { FormattedMessage, useIntl } from 'react-intl';
+
+import { useRouter } from 'next/router';
+
+import { withLocalizedRequests } from 'hoc/locale';
+
+import { useAppSelector } from 'store/hooks';
+import { session } from 'store/session';
+
+import { InferGetStaticPropsType } from 'next';
+
+import useMe from 'hooks/me';
+
+import { loadI18nMessages } from 'helpers/i18n';
+
+import Alert from 'components/alert';
+import Button from 'components/button';
+import ErrorMessage from 'components/forms/error-message';
+import Input from 'components/forms/input';
+import Loading from 'components/loading';
+import { Paths, UserRoles } from 'enums';
+import NakedLayout, { NakedLayoutProps } from 'layouts/naked';
+import { PageComponent } from 'types';
+import { SignIn, SignInCodeForm } from 'types/sign-in';
+
+import { useSignIn } from 'services/authentication/authService';
+
+export const getStaticProps = withLocalizedRequests(async ({ locale }) => {
+  return {
+    props: {
+      intlMessages: await loadI18nMessages({ locale }),
+    },
+  };
+});
+
+type ProjectDeveloperProps = InferGetStaticPropsType<typeof getStaticProps>;
+
+const CODE_LENGTH = 5;
+
+const SignInCode: PageComponent<ProjectDeveloperProps, NakedLayoutProps> = () => {
+  const { formatMessage } = useIntl();
+  const { replace } = useRouter();
+  const signIn = useSignIn();
+  const { user, isLoading: isUserLoading } = useMe();
+
+  // Email and password got from login page
+  const sessionData = useAppSelector(session);
```

Comment:
I understand that you need to share the user's email and password between the `index.tsx` and `code.tsx` page but I wonder if using Redux is the best solution, given this state is not used anywhere else and can be discarded as soon as the user is logged in.

I also wonder if that pauses a security risk that we're storing the password in clear.

Have you thought about using some hidden URL params instead? I believe that we could send the user to `/sign-in/code?email=XXX&password=XXX` and actually hide the params so the URL would look like `/sign-in/code`. I think it can be achieved with the `as` property of Next.js' router.

There are several places where we use this technique in the code already, like here: https://github.com/Vizzuality/heco-invest/blob/63111388539ae3c2c818425626933b75dcac1f32/frontend/containers/dashboard/account-information/public-profile/component.tsx#L75-L80